src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/validappeal/SyaAppealValidator.java

```diff
@@ -0,0 +1,163 @@
+import uk.gov.hmcts.reform.sscs.ccd.validation.sscscasedata.AppealValidator;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.service.DwpAddressLookupService;
+
+@Component
+@Slf4j
+public class SyaAppealValidator extends AppealValidator {
+    private static final String LOGSTR_VALIDATION_ERRORS = "Errors found while validating exception record id {} - {}";
+    private static final String LOGSTR_VALIDATION_WARNING = "Warnings found while validating exception record id {} - {}";
+
+    private final IdamService idamService;
+    private final CcdService ccdService;
+
+    public SyaAppealValidator(IdamService idamService,
+                              CcdService ccdService,
+                              DwpAddressLookupService dwpAddressLookupService,
+                              AddressValidator addressValidator,
+                              List<String> titles) {
+        super(dwpAddressLookupService, addressValidator, SYA_APPEAL, titles);
+        this.idamService = idamService;
+        this.ccdService = ccdService;
+    }
+
+    public PreSubmitCallbackResponse<SscsCaseData> validateAppeal(CaseDetails<SscsCaseData> caseDetails, Map<String, Object> appealData, boolean ignoreMrnValidation) {
+
+        Map<String, List<String>> errsWarns =
+                validateAppeal(new HashMap<>(), appealData, ignoreMrnValidation, false, false);
+
+        CaseResponse caseValidationResponse = CaseResponse.builder()
+                .errors(errsWarns.get("errors"))
+                .warnings(errsWarns.get("warnings"))
+                .transformedCase(appealData)
+                .build();
+
+        PreSubmitCallbackResponse<SscsCaseData> validationErrorResponse = convertWarningsToErrors(caseDetails.getCaseData(), caseValidationResponse); //convertWarn.. can be in tribs.
+
+        if (validationErrorResponse != null) {
+            log.info(LOGSTR_VALIDATION_ERRORS, caseDetails.getId(), ".");
+            return validationErrorResponse;
+        } else {
+            log.info("Appeal {} validated successfully", caseDetails.getId());
+
+            PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(caseDetails.getCaseData());
+
+            if (caseValidationResponse.getWarnings() != null) {
+                preSubmitCallbackResponse.addWarnings(caseValidationResponse.getWarnings());
+            }
+
+            IdamTokens tokens = idamService.getIdamTokens();
+
+            checkForMatches(caseValidationResponse.getTransformedCase(), tokens);
+
+            return preSubmitCallbackResponse;
+        }
+    }
+
+    public Map<String, Object> checkForMatches(Map<String, Object> sscsCaseData, IdamTokens token) {
+        Appeal appeal = (Appeal) sscsCaseData.get("appeal");
+        String nino = "";
+        if (appeal != null && appeal.getAppellant() != null
```

Comment:
is it worth creating a function that returns boolean for this chain of null checks? Could go in util class

Reply:
can probably move to PartyValidator class, since it has a few similar methods